src/tests/reachability/get_leaves_tests.rs

```diff
@@ -0,0 +1,90 @@
+    "Simple nesting")]
+    #[test_case(TransitionID::Composition(
+        Box::new(TransitionID::Conjunction(
+            Box::new(TransitionID::Simple("a".to_string())),
+            Box::new(TransitionID::Composition(
+                Box::new(TransitionID::Simple("b".to_string())),
+                Box::new(TransitionID::Simple("c".to_string()))
+            ))
+        )),
+        Box::new(TransitionID::Composition(
+            Box::new(TransitionID::Simple("d".to_string())),
+            Box::new(TransitionID::Simple("e".to_string()))
+        ))
+    ),
+    vec![
+        vec!(TransitionID::Simple("a".to_string())), 
+        vec!(TransitionID::Simple("b".to_string())), 
+        vec!(TransitionID::Simple("c".to_string())), 
+        vec!(TransitionID::Simple("d".to_string())), 
+        vec!(TransitionID::Simple("e".to_string()))];
+    "Multiple conjunction and composition")]
+    #[test_case(TransitionID::Quotient(
+        vec!(TransitionID::Simple("a".to_string())),
+        vec!(TransitionID::Simple("b".to_string()))
+    ),
+    vec![vec!(TransitionID::Simple("a".to_string())), vec!(TransitionID::Simple("b".to_string()))];
+    "simple quotient")]
+    #[test_case(TransitionID::Quotient(
+        vec!(TransitionID::Simple("a".to_string()), TransitionID::Simple("b".to_string())),
+        vec!(TransitionID::Simple("c".to_string()), TransitionID::Simple("d".to_string()), TransitionID::Simple("e".to_string()))
+    ),
+    vec![
+        vec!(TransitionID::Simple("a".to_string()), TransitionID::Simple("b".to_string())), 
+        vec!(TransitionID::Simple("c".to_string()), TransitionID::Simple("d".to_string()), TransitionID::Simple("e".to_string()))];
+    "quotient with vec")]
+    #[test_case(
+        TransitionID::Conjunction(
+            Box::new(
+                TransitionID::Quotient(
+                    vec![
+                        TransitionID::Conjunction(
+                            Box::new(TransitionID::Simple("a".to_string())),
+                            Box::new(TransitionID::Simple("b".to_string())), 
+                        ),
+                        TransitionID::Conjunction(
+                            Box::new(TransitionID::Simple("c".to_string())),
+                            Box::new(TransitionID::Simple("d".to_string())), 
+                        )
+                    ],
+                    vec![TransitionID::Simple("e".to_string()), TransitionID::Simple("f".to_string())]
+                )
+            ),
+            Box::new(TransitionID::Simple("g".to_string()))
+        ),
+        vec![
+            vec!(TransitionID::Simple("a".to_string()), TransitionID::Simple("c".to_string())), 
+            vec!(TransitionID::Simple("b".to_string()), TransitionID::Simple("d".to_string())),
+            vec!(TransitionID::Simple("e".to_string()), TransitionID::Simple("f".to_string())),
+            vec!(TransitionID::Simple("g".to_string()))];
```

Comment:
Cant these just parse a query instead of doing this manually? It is pretty unwieldy

Reply:
It tests what should be tested